packages/tools/src/tools/annotation/SplineROITool.ts

```diff
@@ -0,0 +1,895 @@
+
+  deleteControlPoint = (
+    evt: EventTypes.InteractionEventType,
+    annotation: SplineROIAnnotation
+  ) => {
+    const splineType = annotation.data.spline.type;
+    const splineConfig = this._getSplineConfig(splineType);
+    const maxDist = splineConfig.delControlPointDistance;
+
+    if (splineConfig.deleteControlPointEnabled === false) {
+      return;
+    }
+
+    const eventDetail = evt.detail;
+    const { element, currentPoints } = eventDetail;
+    const { canvas: canvasPos } = currentPoints;
+    const spline = this._getAnnotationSpline(element, annotation);
+    const closestControlPoint = spline.getClosestControlPointWithinRange(
+      canvasPos,
+      maxDist
+    );
+
+    if (!closestControlPoint) {
+      return;
+    }
+
+    this._deleteControlPointByIndex(
+      element,
+      annotation,
+      closestControlPoint.index
+    );
+  };
+
+  private _deleteControlPointByIndex(
+    element: HTMLDivElement,
+    annotation: SplineROIAnnotation,
+    controlPointIndex: number
+  ) {
+    const enabledElement = getEnabledElement(element);
+    const { points: controlPoints } = annotation.data.handles;
+
+    // There is no curve with only 2 points
+    if (controlPoints.length === 3) {
+      removeAnnotation(annotation.annotationUID);
+    } else {
+      controlPoints.splice(controlPointIndex, 1);
+      this._createOrUpdateAnnotationSpline(element, annotation);
+    }
+
+    const { renderingEngine } = enabledElement;
+    const viewportIdsToRender = getViewportIdsWithToolToRender(
+      element,
+      this.getToolName()
+    );
+
+    triggerAnnotationRenderForViewportIds(renderingEngine, viewportIdsToRender);
+  }
+
+  private _createOrUpdateAnnotationSpline = (
+    element: HTMLDivElement,
+    annotation: SplineROIAnnotation
+  ): ISpline => {
+    const enabledElement = getEnabledElement(element);
+    const { viewport } = enabledElement;
+    const { worldToCanvas } = viewport;
+    let spline = this.splines.get(annotation.annotationUID);
+
+    if (!spline) {
+      const splineType = annotation.data.spline.type;
+      const splineConfig = this._getSplineConfig(splineType);
+
+      console.log('>>>> create :: splineConfig :: ', splineConfig);
+
+      spline = new splineConfig.Class();
+      spline.resolution = splineConfig.resolution;
+
+      if (
+        splineConfig.scale !== undefined &&
+        (<any>spline).scale !== undefined
+      ) {
+        (<any>spline).scale = splineConfig.scale;
+      }
+
+      this.splines.set(annotation.annotationUID, spline);
+    }
+
+    const { data } = annotation;
+    const worldPoints = data.handles.points;
+    const canvasPoints = worldPoints.map(worldToCanvas);
+
+    spline.setControlPoints(canvasPoints);
+    spline.closed = !!data.spline?.closed;
+
+    return spline;
+  };
+
+  private _getAnnotationSpline = (
+    element: HTMLDivElement,
+    annotation: SplineROIAnnotation
+  ) => {
+    let spline = this.splines.get(annotation.annotationUID);
```

Comment:
again not sure what is spline here, is this the one spline (with all segments and control points)? then I"m not sure why it is acting as state manager and has knowledge of other annotationUIDs